src/Agent.Worker/TestResults/Parser.cs

```diff
@@ -72,7 +72,9 @@ public class JUnitParser : Parser, IParser
         protected override ITestResultParser GetTestResultParser(IExecutionContext executionContext)
         {
             var traceListener = new CommandTraceListener(executionContext);
-            return new JUnitResultParser(traceListener);
+            var featureFlagService = executionContext.GetHostContext().GetService<IFeatureFlagService>();
+            var enableJunitAttachments = featureFlagService.GetFeatureFlagState(TestResultsConstants.JUnitTestCaseAttachmentsEnabled, TestResultsConstants.TCMServiceInstanceGuid);
```

Comment:
This may be the first time we get a FF from something other than TFS service (at a glance it appears to be).  Not necessarily an issue, but please double-check the REST permissions.  It might be better to just define the FF in TFS service to avoid deployment confusion if this is the only one.

Reply:
It looks like TestDataPublisher already uses Feature Flags from TCM service so I think this should be okay. I would prefer to keep this FF in TCM since that would be consistent with the implementation in the EXE task implementation used for Windows agents.

https://github.com/microsoft/azure-pipelines-agent/blob/f87750b36c64c82400cb53791975e8f0f50fcd77/src/Agent.Worker/TestResults/TestDataPublisher.cs#L97